### PR TITLE
1.add MCU options display when use -v flag

### PR DESCRIPTION
**Pull request by trlsmax**
1.add MCU options display when use -v flag
        MCU options data is port from kstc89/12isp program
2.add -e flag to enable erase EEPROM data on next download

just learn very little python programing today, hope don't make too much trouble and mistake
tested on STC89C58RD+

### CODE REVIEW

File: stcflash.py [hunk 1]
```diff
@@ -422,12 +423,41 @@ def program(prog, code):
...
+            logging.info("[%c] %s"
+                         % ('X' if prog.info[8] & key != 0 else ' ', desc))
+
+        switchs4 = {0x02: "Not erase EEPROM data",
+                    0x01: "Download regardless of P1"}
+        for key, desc in switchs4.items():
+            logging.info("[%c] %s"
+                         % ('X' if prog.info[10] & key != 0 else ' ', desc))
+
+        logging.info("WDT prescal is %d" % math.pow(2,(prog.info[8] & 0x07) + 1))
```
Comment (laborer) [inline stcflash.py]: You can use `2**n` instead.  So there is no need to `import math`.

File: stcflash.py [hunk 2]
```diff
@@ -422,12 +423,41 @@ def program(prog, code):
...
+                    0x04: "WDT count in idle mode"}
+        for key, desc in switchs3.items():
+            logging.info("[%c] %s"
+                         % ('X' if prog.info[8] & key != 0 else ' ', desc))
+
+        switchs4 = {0x02: "Not erase EEPROM data",
+                    0x01: "Download regardless of P1"}
+        for key, desc in switchs4.items():
+            logging.info("[%c] %s"
+                         % ('X' if prog.info[10] & key != 0 else ' ', desc))
```
Comment (laborer) [inline stcflash.py]: The above 4 for loops can be merged into one, like this,
```python
switch = {(6, 0x40, "Disable reset2 low level detect"),
          (6, 0x01, "Reset pin not use as I/O port")}
for pos, bit, desc in switch:
    logging.info("[%c] %s"
                 % ('X' if prog.info[pos] & bit != 0 else ' ', desc))
```

File: stcflash.py [hunk 3]
```diff
@@ -404,7 +405,7 @@ def autoisp(conn, baud, magic):
     time.sleep(0.5)
     conn.baudrate = bak
 
-def program(prog, code):
+def program(prog, code, erase_eeprom=None):
```
Comment (laborer) [inline stcflash.py]: Use `erase_eeprom=True`

**Comment by laborer** [comment]
Thanks for the contribution.  You code looks great.  I will merge the commit, but please address the issues I noted in the code.

**Comment by trlsmax** [comment]
Hi
Just commit some change which fix the issues you point out, please review the code.

**Comment by laborer** [comment]
Thanks!  I will test your code.

**Pull request merged**